agent/xds/clusters.go

```diff
@@ -1427,10 +1459,6 @@ func (s *ResourceGenerator) makeExportedUpstreamClustersForMeshGateway(cfgSnap *
 
 // injectSANMatcher updates a TLS context so that it verifies the upstream SAN.
 func injectSANMatcher(tlsContext *envoy_tls_v3.CommonTlsContext, matchStrings ...string) error {
-	if tlsContext == nil {
-		return fmt.Errorf("invalid type: expected CommonTlsContext_ValidationContext not to be nil")
-	}
-
```

Comment:
Was this an unintentional change? I don't see this function getting called in this PR, so I'm suspecting so

Reply:
Yeah, we should add this back in. I don't fully trust the xds/proxycfg packages around their panic handling, and I wouldn't like to unintentionally add in some places where we panic and crash Consul rather than just spamming the logs with errors.

Reply:
I do think we removed it intentionally because it wasn't getting initialized, at least when we were working on this chunk, but I'll test it.

Reply:
Restored